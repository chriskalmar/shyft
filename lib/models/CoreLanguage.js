
import {
  dataTypes,
  Entity,
} from '../../';


const {
  DataTypeString,
} = dataTypes



export const CoreLanguage = new Entity({
  name: 'language',
  domain: 'core',
  description: 'A language',

  // indexing: {
  //   unique: [
  //     [ 'name' ],
  //     [ 'iso_code' ],

  attributes: {

    name: {
      type: DataTypeString,
      description: 'The name of the language',
      maxLength: 30,
      required: true
    },

    nativeName: {
      type: DataTypeString,
      description: 'The native name of the language',
      maxLength: 30,
      required: true
    },

    isoCode: {
      type: DataTypeString,
      description: 'ISO code of the language',
      pattern: '^[a-z]+$',
      minLength: 2,
      maxLength: 2,
      required: true
    },
  }

})

