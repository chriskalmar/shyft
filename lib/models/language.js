
'use script';

module.exports = {

  domain: 'core',

  entities: [

    {
      name: 'language',
      description: 'a language',

      cascadeOnDelete: true,

      attributes: [

        {
          name: 'name',
          type: 'string',
          description: 'Name',
          maxLength: 30,
          required: true
        },

        {
          name: 'native_name',
          type: 'string',
          description: 'Name',
          maxLength: 30,
          required: true
        },

        {
          name: 'iso_code',
          type: 'string',
          description: 'ISO 639-1 code of the language',
          maxLength: 2,
          required: true
        },

      ],

      indexing: {
        unique: [
          [ 'name' ],
          [ 'iso_code' ],
        ]
      }

    }

  ]

}
