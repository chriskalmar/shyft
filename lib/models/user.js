
'use script';

module.exports = {

  domain: 'core',

  entities: [

    {
      name: 'user',
      description: 'a user',

      isUserTable: true,
      cascadeOnDelete: true,

      attributes: [

        {
          name: 'username',
          type: 'string',
          description: 'Username',
          pattern: '^(?![_.])(?!.*[_.]{2})[a-zA-Z0-9._]+$',
          minLength: 3,
          maxLength: 30,
          required: true
        },

        {
          name: 'first_name',
          type: 'string',
          description: 'First name',
          maxLength: 30
        },

        {
          name: 'last_name',
          type: 'string',
          description: 'Last name',
          maxLength: 30
        },

        {
          name: 'language',
          type: 'reference',
          description: 'User language',
          target: 'core::language',
          required: true
        },

        {
          name: 'email',
          type: 'email',
          description: 'Email address',
          required: true
        },

        {
          name: 'password',
          type: 'password',
          description: 'User password',
          required: true
        }
      ],

      indexing: {
        unique: [
          [ 'username' ],
          [ 'email' ],
        ]
      }

    }

  ]

}
