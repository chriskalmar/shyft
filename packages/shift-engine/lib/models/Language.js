
import Entity from '../entity/Entity';
import { DataTypeString } from '../datatype/dataTypes';
import Index, { INDEX_UNIQUE } from '../index/Index';


export const Language = new Entity({
  name: 'Language',
  description: 'A language',

  indexes: [
    new Index({
      type: INDEX_UNIQUE,
      attributes: [ 'name' ],
    }),
    new Index({
      type: INDEX_UNIQUE,
      attributes: [ 'isoCode' ],
    }),
  ],

  attributes: {

    name: {
      type: DataTypeString,
      description: 'The name of the language',
      required: true
    },

    nativeName: {
      type: DataTypeString,
      description: 'The native name of the language',
      required: true
    },

    isoCode: {
      type: DataTypeString,
      description: 'ISO code of the language',
      required: true
    },
  }

})

