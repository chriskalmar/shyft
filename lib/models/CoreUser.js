
import {
  datatypes,
  Entity,
} from '../../';

import CoreLanguage from './CoreLanguage';

const {
  DataTypeString,
} = datatypes

export const CoreUser = new Entity({
  name: 'user',
  domain: 'core',
  description: 'A user',

  // isUserTable: true,
  // sequenceGenerator: 'timebased',

  // indexing: {
  //   unique: [
  //     [ 'username' ],
  //     [ 'email' ],


  attributes: {

    username: {
      type: DataTypeString,
      description: 'Username',
      pattern: '^(?![_.])(?!.*[_.]{2})[a-zA-Z0-9._]+$',
      minLength: 3,
      maxLength: 30,
      required: true
    },

    firstName: {
      type: DataTypeString,
      description: 'First name',
      maxLength: 30
    },

    lastName: {
      type: DataTypeString,
      description: 'Last name',
      maxLength: 30
    },

    language: {
      type: CoreLanguage,
      description: 'User language',
      required: true
    },

    email: {
      type: DataTypeString,
      description: 'Email address',
      required: true
    },

    password: {
      type: DataTypeString,
      description: 'User password',
      required: true
    }
  }

})



