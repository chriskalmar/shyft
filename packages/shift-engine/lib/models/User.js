
import Entity from '../entity/Entity';
import { DataTypeString } from '../datatype/dataTypes';
import Index, { INDEX_UNIQUE } from '../index/Index';

import { Language } from './Language';


export const User = new Entity({
  name: 'User',
  domain: 'core',
  description: 'A user',

  // isUserTable: true,
  // sequenceGenerator: 'timebased',

  indexing: [
    new Index({
      type: INDEX_UNIQUE,
      attributes: [ 'username' ],
    }),
    new Index({
      type: INDEX_UNIQUE,
      attributes: [ 'email' ],
    }),
  ],


  attributes: {

    username: {
      type: DataTypeString,
      description: 'Username',
      pattern: '^(?![_.])(?!.*[_.]{2})[a-zA-Z0-9._]+$',
      minLength: 3,
      maxLength: 30,
      required: true
    },

    firstName: {
      type: DataTypeString,
      description: 'First name',
      maxLength: 30
    },

    lastName: {
      type: DataTypeString,
      description: 'Last name',
      maxLength: 30
    },

    language: {
      type: Language,
      description: 'User language',
      required: true
    },

    email: {
      type: DataTypeString,
      description: 'Email address',
      required: true
    },

    password: {
      type: DataTypeString,
      description: 'User password',
      required: true
    }
  }

})



