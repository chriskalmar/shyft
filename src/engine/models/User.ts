import { Entity } from '../entity/Entity';
import { DataTypeString } from '../datatype/dataTypes';
import { Index, INDEX_UNIQUE } from '../index/Index';

import { Language } from './Language';

export const User = new Entity({
  name: 'User',
  description: 'A user',

  // isUserTable: true,
  // sequenceGenerator: 'timebased',

  indexes: [
    new Index({
      type: INDEX_UNIQUE,
      attributes: [ 'username' ],
    }),
    new Index({
      type: INDEX_UNIQUE,
      attributes: [ 'email' ],
    }),
  ],

  attributes: {
    username: {
      type: DataTypeString,
      description: 'Username',
      required: true,
    },

    firstName: {
      type: DataTypeString,
      description: 'First name',
    },

    lastName: {
      type: DataTypeString,
      description: 'Last name',
    },

    language: {
      type: Language,
      description: 'User language',
      required: true,
    },

    email: {
      type: DataTypeString,
      description: 'Email address',
      required: true,
    },

    password: {
      type: DataTypeString,
      description: 'User password',
      required: true,
    },
  },
});
