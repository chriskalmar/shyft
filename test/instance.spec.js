import './beforeAll';
import {
  findOne,
  findOneByValue,
} from './db';

import {
  asAdmin,
  removeDynamicData,
} from './testUtils';


import { Profile } from './models/Profile';
import { Board } from './models/Board';
import { Participant } from './models/Participant';


describe('postgres', () => {

  it('should fetch instances by ID', async () => {
    let profile

    profile = await findOne(Profile, 10, {}, asAdmin())
    expect(removeDynamicData(Profile, profile)).toMatchSnapshot()

    profile = await findOne(Profile, 20, {}, asAdmin())
    expect(removeDynamicData(Profile, profile)).toMatchSnapshot()

    profile = await findOne(Profile, 30, {}, asAdmin())
    expect(removeDynamicData(Profile, profile)).toMatchSnapshot()


    let board

    board = await findOne(Board, 10, {}, asAdmin())
    expect(removeDynamicData(Board, board)).toMatchSnapshot()

    board = await findOne(Board, 20, {}, asAdmin())
    expect(removeDynamicData(Board, board)).toMatchSnapshot()

    board = await findOne(Board, 30, {}, asAdmin())
    expect(removeDynamicData(Board, board)).toMatchSnapshot()


    let participant

    participant = await findOne(Participant, 200, {}, asAdmin())
    expect(removeDynamicData(Participant, participant)).toMatchSnapshot()

    participant = await findOne(Participant, 201, {}, asAdmin())
    expect(removeDynamicData(Participant, participant)).toMatchSnapshot()

    participant = await findOne(Participant, 202, {}, asAdmin())
    expect(removeDynamicData(Participant, participant)).toMatchSnapshot()
  })


  it('should fetch instances by value', async () => {
    let profile

    profile = await findOneByValue(Profile, { username: '----unknown----' }, asAdmin())
    expect(profile).toBeUndefined()

    profile = await findOneByValue(Profile, { username: 'katrina560' }, asAdmin())
    expect(removeDynamicData(Profile, profile)).toMatchSnapshot()

    profile = await findOneByValue(Profile, { username: 'clark218' }, asAdmin())
    expect(removeDynamicData(Profile, profile)).toMatchSnapshot()

    profile = await findOneByValue(Profile, { }, asAdmin())
    expect(removeDynamicData(Profile, profile)).toMatchSnapshot()

    let participant

    participant = await findOneByValue(Participant, { board: 9999 }, asAdmin())
    expect(participant).toBeUndefined()

    participant = await findOneByValue(Participant, { board: 9, invitee: 7, inviter: 52 }, asAdmin())
    expect(removeDynamicData(Participant, participant)).toMatchSnapshot()

    participant = await findOneByValue(Participant, { }, asAdmin())
    expect(removeDynamicData(Participant, participant)).toMatchSnapshot()

  })


  it('should return null when instance is not found', async () => {
    const profile = await findOne(Profile, 10000, {}, asAdmin())
    expect(profile).toBeNull()
  })

})
