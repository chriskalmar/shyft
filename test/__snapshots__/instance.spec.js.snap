// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postgres should fetch instances by ID 1`] = `
Object {
  "confirmedAt": null,
  "id": "10",
  "password": "16149626b0365e04a9a52d02e17ca8aeb63680b9f5ee2ba334b0b1a38a833cfe",
  "username": "leon807",
}
`;

exports[`postgres should fetch instances by ID 2`] = `
Object {
  "confirmedAt": null,
  "id": "20",
  "password": "bcee72e6df5d56319125920f6c69d8001e938e122cd14597e547d49cc9126e42",
  "username": "katrina560",
}
`;

exports[`postgres should fetch instances by ID 3`] = `
Object {
  "confirmedAt": null,
  "id": "30",
  "password": "c6520201ad739678e8e86c983c072c1fd95dba3b98d5023e40a8d895924b429f",
  "username": "carissa722",
}
`;

exports[`postgres should fetch instances by ID 4`] = `
Object {
  "createdBy": "71",
  "id": "10",
  "isPrivate": true,
  "name": "Sit aut",
  "owner": "71",
  "updatedBy": "71",
}
`;

exports[`postgres should fetch instances by ID 5`] = `
Object {
  "createdBy": "10",
  "id": "20",
  "isPrivate": false,
  "name": "Sunt quo",
  "owner": "10",
  "updatedBy": "10",
}
`;

exports[`postgres should fetch instances by ID 6`] = `
Object {
  "createdBy": "6",
  "id": "30",
  "isPrivate": false,
  "name": "Fuga accusantium voluptatem",
  "owner": "6",
  "updatedBy": "6",
}
`;

exports[`postgres should fetch instances by ID 7`] = `
Object {
  "board": "21",
  "id": "200",
  "invitee": "107",
  "inviter": "107",
  "state": 50,
}
`;

exports[`postgres should fetch instances by ID 8`] = `
Object {
  "board": "5",
  "id": "201",
  "invitee": "101",
  "inviter": "55",
  "state": 10,
}
`;

exports[`postgres should fetch instances by ID 9`] = `
Object {
  "board": "9",
  "id": "202",
  "invitee": "7",
  "inviter": "52",
  "state": 20,
}
`;

exports[`postgres should fetch instances by value 1`] = `
Object {
  "confirmedAt": null,
  "id": "20",
  "password": "bcee72e6df5d56319125920f6c69d8001e938e122cd14597e547d49cc9126e42",
  "username": "katrina560",
}
`;

exports[`postgres should fetch instances by value 2`] = `
Object {
  "confirmedAt": null,
  "id": "14",
  "password": "87ba65bd3e47783927b0035d5537c2bfc92362955d3a6fe39289ab03075b5b7c",
  "username": "clark218",
}
`;

exports[`postgres should fetch instances by value 3`] = `
Object {
  "confirmedAt": null,
  "id": "1",
  "password": "8fed8061e1e33d96169b9026605172a8fe21eb1ce51756187195be0330536fb7",
  "username": "annabel711",
}
`;

exports[`postgres should fetch instances by value 4`] = `
Object {
  "board": "9",
  "id": "202",
  "invitee": "7",
  "inviter": "52",
  "state": 20,
}
`;

exports[`postgres should fetch instances by value 5`] = `
Object {
  "board": "41",
  "id": "1",
  "invitee": "95",
  "inviter": "95",
  "state": 50,
}
`;

exports[`postgres should utilize data loader 1`] = `
Array [
  Object {
    "confirmedAt": null,
    "id": "10",
    "password": "16149626b0365e04a9a52d02e17ca8aeb63680b9f5ee2ba334b0b1a38a833cfe",
    "username": "leon807",
  },
  Object {
    "confirmedAt": null,
    "id": "20",
    "password": "bcee72e6df5d56319125920f6c69d8001e938e122cd14597e547d49cc9126e42",
    "username": "katrina560",
  },
  Object {
    "confirmedAt": null,
    "id": "30",
    "password": "c6520201ad739678e8e86c983c072c1fd95dba3b98d5023e40a8d895924b429f",
    "username": "carissa722",
  },
  Object {
    "confirmedAt": null,
    "id": "20",
    "password": "bcee72e6df5d56319125920f6c69d8001e938e122cd14597e547d49cc9126e42",
    "username": "katrina560",
  },
  Object {
    "confirmedAt": null,
    "id": "10",
    "password": "16149626b0365e04a9a52d02e17ca8aeb63680b9f5ee2ba334b0b1a38a833cfe",
    "username": "leon807",
  },
]
`;
